chainntnfs/txconfnotifier.go

```diff
@@ -225,14 +225,37 @@ func (tcn *TxConfNotifier) Register(ntfn *ConfNtfn) (bool, uint32, error) {
 
 	confSet.ntfns[ntfn.ConfID] = ntfn
 
-	switch {
-	case ntfn.HeightHint > tcn.currentHeight:
-		return false, 0, nil
-	default:
-		confSet.rescanCount++
-		return true, tcn.currentHeight, nil
+	switch confSet.rescanStatus {
+
+	// A prior rescan has already completed and we are actively watching at
+	// tip for this txid.
+	case rescanComplete:
+		return nil, nil
+
+	// A rescan is already in progress, return here to prevent dispatching
+	// another. When the scan returns, this notifications details will be
+	// updated as well.
+	case rescanPending:
+		return nil, nil
+
+	// If no rescan has been dispatched, attempt to do so now.
+	case rescanNotStarted:
+	}
+
+	// If the provided or cached height hint indicates that the transaction
+	// is to be confirmed at a height greater than the conf notifier's
+	// current height, we'll refrain from spawning a historical dispatch.
+	if startHeight > tcn.currentHeight {
+		// Set the rescan status to complete, which will allow the conf
+		// notifier to start delivering messages for this set
+		// immediately.
+		confSet.rescanStatus = rescanComplete
```

Comment:
Yes I believe that's correct, only the height hint of the first registration would be considered in that case. The alternative is to do some ARC magic and actually maintain a count of all active rescans, at the expense of (probably) rescanning unnecessarily in most instances. This was the approach taken early on, though I agree it'd be better in the long run to just ensure we are passing them in properly instead of being wasteful in manual rescans.

Reply:
it technically would be possible for the confset to track the height hints it's seen so that we can be alerted of the situation, but maybe something to leave to a later PR

Reply:
fixed